### ISSUE TITLE
Unexpected TypeError on NODE_ENV=production in mapValueError

### ISSUE THREAD

**Issue by nxht** [elysiajs/elysia#787]
### What version of Elysia is running?

1.1.6, 1.1.7

### What platform is your computer?

Linux 5.15.153.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

```ts
import Elysia, { t } from 'elysia';

export const app = new Elysia()
  .get('/', () => 'hello world', {
    query: t.Object({
      name: t.String(),
    }),
  })
  .listen(3000);
```

Run in production

```bash
NODE_ENV=production bun src/index.ts                                                                                                                    
```

Then access http://localhost:3000/

### What is the expected behavior?

ValidationError. Like in other NODE_ENV case but perhaps less detailed

```json
{
  "type": "validation",
  "on": "query",
  "summary": "Property 'name' is missing",
  "property": "/name",
  "message": "Required property",
  "expected": {
    "name": ""
  },
  "found": {
    
  },
  "errors": [
    {
      "type": 45,
      "schema": {
        "type": "string"
      },
      "path": "/name",
      "message": "Required property",
      "summary": "Property 'name' is missing"
    },
    {
      "type": 54,
      "schema": {
        "type": "string"
      },
      "path": "/name",
      "message": "Expected string",
      "summary": "Expected  property 'name' to be  string but found: undefined"
    }
  ]
}
```

### What do you see instead?

```json
{
  "name": "TypeError",
  "message": "Right side of assignment cannot be destructured"
}
```

When catch and printed
```
101 | }
102 | 
103 | export const mapValueError = (error: ValueError) => {
104 |   const { message, path, value, type } = error
105 | 
106 |   const property = path.slice(1).replaceAll('/', '.')
                        ^
TypeError: Right side of assignment cannot be destructured
      at s (/home/wsladmin/typescript/test-elysia-simple/node_modules/elysia/src/error.ts:106:19)
      at new T (/home/wsladmin/typescript/test-elysia-simple/node_modules/elysia/src/error.ts:216:14)
      at handle (79:39)
      at map (62:25)
```

### Additional information

It seems to be coming from `ValidationError` constructor.

const `error` becomes undefined in production case thus causes `customError` to be undefined
And to generate `message` as it's `isProduction`, it runs `mapValueError(error).summary` 
But as `error` is undefined, this causes `mapValueError` to fail

### Have you try removing the `node_modules` and `bun.lockb` and try again yet?

Yes

**Comment by bogeychan** [comment]
Duplicate of https://github.com/elysiajs/elysia/issues/789

**Comment by SaltyAom** [comment]
Should have been fixed with https://github.com/elysiajs/elysia/commit/5f9370c6489f8a1a62d2e2107ec61d0289174443, with test case included, published under 1.1.8.

Let me know if this fix your issue, thanks.

**Comment by nxht** [comment]
The issue seems to be solved in 1.1.8
